contrib/ci/check-dep-versions.py

```
@@ -0,0 +1,303 @@
+        local_meson = os.path.join(repo_dir, "venv", "bin", "meson")
+        if os.path.isfile(local_meson) and os.access(local_meson, os.X_OK):
+            meson_cmd = local_meson
+        else:
+            raise RuntimeError("`meson` not found in PATH or at venv/bin/meson")
+
+    errors: List[str] = []
+    for candidate in _candidate_build_dirs(repo_dir, build_dir):
+        if not os.path.isdir(candidate):
+            continue
+        cmd = [meson_cmd, "introspect", candidate, "--dependencies"]
+        proc = subprocess.run(cmd, capture_output=True, text=True, check=False)
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)